controllers/managedocs_controller.go

```diff
@@ -570,8 +588,14 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 		if err := r.reconcilePrometheusProxyNetworkPolicy(); err != nil {
 			return ctrl.Result{}, err
 		}
-		if err := r.reconcileEgressNetworkPolicy(); err != nil {
-			return ctrl.Result{}, err
+		if r.NetworkType == OVNNetworkType {
+			if err := r.reconcileEgressFirewall(); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			if err := r.reconcileEgressNetworkPolicy(); err != nil {
+				return ctrl.Result{}, err
+			}
 		}
```

Comment:
Please move the conditional into the reconcile function as an early exist conditional and then just call both functions. For example, at the start of `reconcileEgressFirewall` you will have
```
if r.NetworkType != OVNNetworkType {
    return nil
}
```
Same for the other reconcile func.

Please note we are using a similar pattern for deployment types.

Reply:
Done